src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java

```diff
@@ -479,6 +463,13 @@ public boolean isAnonymous() {
 	}
 
 	@Override
+	public boolean isEnclosingType() {
+		final Pattern pattern = Pattern.compile("^([0-9]+)([a-zA-Z]+)$");
```

Comment:
> for sake of future maintenance, explain what this means?

Do you want a comment which explains that the name of a local type starts with a number and end with a letter?

> add a pointer to a document stating this convention for compilers? (or at least for JDT)?

There is no pointer for that. It doesn't exist a fully qualified name for local types ([Oracle doc](https://docs.oracle.com/javase/specs/jls/se7/html/jls-6.html#jls-6.7) and [JDT doc](http://help.eclipse.org/juno/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fcore%2Fdom%2FITypeBinding.html)) but both have an identifier to identify the right local type.